src/parsing_astable.jl

```diff
@@ -0,0 +1,95 @@
+function conditionally_add_symbols!(inputs_to_function, lhs_assignments, col)
+    # if it's already been assigned at top-level,
+    # don't add it to the inputs
+    if haskey(lhs_assignments, col)
+        return lhs_assignments[col]
+    else
+        return addkey!(inputs_to_function, col)
+    end
```

Comment:
```suggestion
    return get!(lhs_assignments, col) do
        gensym()
    end
```

Reply:
Ah this isn't quite right. Notice the two different dicts used, `lhs_assignments` and `inputs_to_function`.